scripts/perf-activation-session.py

```
@@ -592,30 +623,41 @@ def apply_budgets(self) -> None:
+        warnings: list[str] = []
+        # Wall-clock timing budgets are load-sensitive even after best-of-N
+        # sampling, so they are advisory by default (performance is measured,
+        # not gated on a shared CI runner). Deterministic structural budgets
+        # below stay blocking. Pass --fail-on-timing-budget to hard-gate timing.
+        timing_blocking = self.args.fail_on_timing_budget
 
-        def max_budget(label: str, actual: float | int | None, budget: float | int) -> None:
+        def record(blocking: bool, message: str) -> None:
+            (failures if blocking else warnings).append(message)
+
+        def max_budget(label: str, actual: float | int | None, budget: float | int, blocking: bool = True) -> None:
```

Ruff (0.15.17)

[warning] Boolean-typed positional argument in function definition

(FBT001)